mp/src/game/client/hud.cpp

```diff
@@ -520,10 +524,25 @@ void CHud::Shutdown( void )
 
 	// Deleting hudlist items can result in them being removed from the same hudlist (m_bNeedsRemove).
 	//	So go through and kill the last item until the array is empty.
+#ifdef NEO
+	// NEO JANK (nullsystem): The original while/delete loop didn't even make sense as the size didn't really come
```

Comment:
I think the original code's `delete` calls the `~CHudElement` desctructor, which in turn calls `CHud::RemoveHudElement` for any `m_bNeedsRemove` elements, and that's what eventually decrements the `while` counter, and shifts the array left.

Perhaps to play nice with the `CHudElementHelper::CreateAllElements` system, we could declare our HUD with `DECLARE_HUDELEMENT` in the implementation, so we didn't need to manually manage this memory?

Reply:
Looks like it works out. For now I'll apply this `DECLARE_HUDELEMENT` to this, then the rest of the stuff migrating to that as a separate PR.